cmd/inspector/main.go

```diff
@@ -0,0 +1,139 @@
+// Copyright (C) 2018 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package main
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/spf13/cobra"
+	"github.com/zeebo/errs"
+
+	"storj.io/storj/pkg/node"
+	"storj.io/storj/pkg/pb"
+	"storj.io/storj/pkg/process"
+	"storj.io/storj/pkg/provider"
+	"storj.io/storj/pkg/transport"
+)
+
+var (
+	addr = "127.0.0.1:7778"
+	// ErrInspectorDial throws when there are errors dialing the inspector server
+	ErrInspectorDial = errs.Class("error dialing inspector server:")
+	rootCmd          = &cobra.Command{
+		Use:   "inspector",
+		Short: "CLI for interacting with Storj Kademlia network",
+	}
+	countNodeCmd = &cobra.Command{
+		Use:   "count",
+		Short: "count nodes in kademlia and overlay",
+		RunE:  CountNodes,
+	}
+	getBucketsCmd = &cobra.Command{
+		Use:   "list-buckets",
+		Short: "get all buckets in overlay",
+		RunE:  GetBuckets,
+	}
+	getBucketCmd = &cobra.Command{
+		Use:   "ls <bucket_id>",
+		Short: "get all nodes in bucket",
+		RunE:  GetBucket,
+	}
+)
+
+// Inspector gives access to kademlia and overlay cache
+type Inspector struct {
+	identity *provider.FullIdentity
+	client   pb.InspectorClient
+	ctx      context.Context
```

Comment:
Because I used it in the gRPC client. Should I just create a new context.Background() in each method? I figured it was better to use a single context that came from the NewInspector() function. What do you think?

Reply:
I removed context from the struct. 👍